api/service/discovery/service.go

```diff
@@ -85,5 +85,7 @@ func (s *Service) pingPeer(peer p2p.Peer) {
 	content := host.ConstructP2pMessage(byte(0), buffer)
 	s.Host.SendMessage(peer, content)
 	log.Debug("Sent Ping Message to", "peer", peer)
-	s.stakingChan <- peer
+	if s.stakingChan != nil {
+		s.stakingChan <- peer
```

Comment:
What does pingPeer do? My understanding is `peerDiscovery` returns few nodes I can connect to. Is the `peer` in the `pingPeer` function the the `bootnode` I am pinging? or it is one of peer(s) that has already been returned to me by `bootnode` (Or possibly just a test function).

Reply:
pingPeer is to establish the link between A and B.  A got the info of B from bootnode, but A need to send a ping message to B, to establish the link.  Otherwise, B won't know A as well.